src/Discord/WebSockets/Events/MessagePollVoteAdd.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+/*
+ * This file is a part of the DiscordPHP project.
+ *
+ * Copyright (c) 2015-present David Cole <david.cole1340@gmail.com>
+ *
+ * This file is subject to the MIT license that is bundled
+ * with this source code in the LICENSE.md file.
+ */
+
+namespace Discord\WebSockets\Events;
+
+use Discord\Parts\Channel\Poll\PollAnswer;
+use Discord\WebSockets\Event;
+use Discord\Parts\Channel\Channel;
+use Discord\Parts\Channel\Message;
+use Discord\Parts\Channel\Reaction;
+use Discord\Parts\Guild\Guild;
+use Discord\Parts\Thread\Thread;
+
+/**
+ * @link https://discord.com/developers/docs/topics/gateway-events#message-poll-vote-add-message-poll-vote-add-fields
+ *
+ * @since 10.0.0
+ */
+class MessagePollVoteAdd extends Event
+{
+    /**
+     * {@inheritDoc}
+     */
+    public function handle($data)
+    {
+        $guild = null;
+
+        /** @var ?Guild */
+        if (isset($data->guild_id) && $guild = yield $this->discord->guilds->cacheGet($data->guild_id)) {
+            /** @var ?Channel */
+            if (! $channel = yield $guild->channels->cacheGet($data->channel_id)) {
+                /** @var Channel */
+                foreach ($guild->channels as $channel) {
+                    /** @var ?Thread */
+                    if ($thread = yield $channel->threads->cacheGet($data->channel_id)) {
+                        $channel = $thread;
+                        break;
+                    }
+                }
+            }
+        } else {
+            /** @var ?Channel */
+            $channel = yield $this->discord->private_channels->cacheGet($data->channel_id);
+        }
+
+        $answer = new PollAnswer($this->discord, (array) $data, true);
+
+        /** @var ?Message */
+        if (isset($channel) && $message = yield $channel->messages->cacheGet($data->message_id)) {
```

Comment:
need to test more but I don't think the vote message is being able to get fetched from the cache for some reason so this is always `false`

Reply:
this is good to go, i just had `storeMessages` disabled by accident.